### PR TITLE
EPL-6511: Initial Workiva json_schema release

### PR DESCRIPTION
**Pull request by michaelcarter-wf**
## Ultimate problem:
- Build tools were not Workiva standard
- Some dead code was in the repo
- Some analyzer warning were present
- Strong mode was not enabled
- There were no github templates
- We weren't tracking dependencies the workiva way
- The pubspec didn't reflect new ownership of the repo
- There were unused deps in the pubspec

## How it was fixed:
- do all the things...

## Testing suggestions:
- See the new build passes and read the logs (it now verifies formatting, coverage, tests, unused deps)

## Potential areas of regression:
- Builds?

### CODE REVIEW

**Comment by aviary-wf** [comment]
## Raven
<details>
<summary>Number of Findings: 0</summary>

</details>

**Comment by michaelcarter-wf** [comment]
@patefacio would love your thoughts here too if you're interested in reviewing, but don't feel you must. Thanks!

**Comment by patefacio** [comment]
@michaelcarter-wf Looks good. Thanks for doing this.

File: test/unit/vm/json_schema/validation_test.dart [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    Logger.root.level = Level.OFF;
+  }
+// custom <main>
+
+  ////////////////////////////////////////////////////////////////////////
+  // Uncomment to see logging of excpetions
+  // Logger.root.onRecord.listen((LogRecord r) =>
+  //   print("${r.loggerName} [${r.level}]:\t${r.message}"));
+
+  Logger.root.level = Level.OFF;
```
Comment (jamesromo-wf) [inline test/unit/vm/json_schema/validation_test.dart]: Do we want to turn this logger on and also log the exceptions above?

File: smithy.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+project: dart
+language: dart
+
+# dart 1.24.2
+runner_image: drydock-prod.workiva.net/workiva/smithy-runner-generator:179735
```
Comment (jamesromo-wf) [inline smithy.yaml]: Do we have smithy turned on for this repo?

File: pubspec.yaml [hunk 3]
```diff
@@ -1,39 +1,24 @@
 name: json_schema
 version: 1.0.7
```
Comment (jamesromo-wf) [inline pubspec.yaml]: Do we want our "initial Workiva release" version? Edit: looks like the Changelog has it at 1.0.8

File: example/from_json/movie_sample.dart [hunk 4]
```diff
@@ -13,54 +50,44 @@ main() {
...
-      } else {
-        print('$movies:\nvalidates!');
-      }
-    });
-
+  Schema.createSchema(movieSchema).then((schema) {
+    var validator = new Validator(schema);
+    bool validates = validator.validate(movies);
+    if (!validates) {
+      print("Errors: ${validator.errors}");
```
Comment (jamesromo-wf) [inline example/from_json/movie_sample.dart]: I would imagine we'd want to change these prints to go through the logger, but that can be saved for another time.

File: smithy.yaml (hunk 2 above)
Comment (michaelcarter-wf) [reply smithy.yaml]: dependency tracking for Rosie

Comment (michaelcarter-wf) [reply smithy.yaml]: ah, ya, asked smithy people about that on friday and didn't get any response. I'll pester them again...

Comment (michaelcarter-wf) [reply smithy.yaml]: Waiting on ATLAS-23057

File: example/from_json/movie_sample.dart (hunk 4 above)
Comment (michaelcarter-wf) [reply example/from_json/movie_sample.dart]: might do it if the smithy setup takes a while anyway

File: pubspec.yaml (hunk 3 above)
Comment (michaelcarter-wf) [reply pubspec.yaml]: Hoping MARV will bump this

File: test/unit/vm/json_schema/validation_test.dart (hunk 1 above)
Comment (michaelcarter-wf) [reply test/unit/vm/json_schema/validation_test.dart]: I'll take care of this with other logging stuff

File: smithy.yaml (hunk 2 above)
Comment (michaelcarter-wf) [reply smithy.yaml]: done

File: example/from_json/movie_sample.dart (hunk 4 above)
Comment (michaelcarter-wf) [reply example/from_json/movie_sample.dart]: saving this for a logging cleanup ticket: EPL-6664

**Comment by bryanrezende-wf** [comment]
+10 Everything looks good. Everything passes and logs are present in Travis

**Comment by michaelcarter-wf** [comment]
@Workiva/release-management-pp

**Pull request merged**